DietPi-Software | Fix Tor relay configuration

nonnorm opened the pull request:
<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->
**Status**: Ready

**Commit list/description**:
<!--
- DietPi-Config | Add "Fan control" option to "Performance Options"
![Screenshot](https://xxx.github.com/images/xxx.png)
- DietPi-Config | Add "Fan control" support for Odroid C2
- DietPi-Config | Syntax fix
-->
- DietPi-Software | Fix IPv6 and obfs4 configuration
- DietPi-Software | CodeFactor Fixes
- DietPi-Software | Remove random indent
- DietPi-Software | Change RegEx
- DietPi-Software | Add INPUTBOX
- DietPi-Software | Move IPv6Exit

nonnorm committed:
Fix IPv6 and obfs4 configuration

nonnorm committed:
CodeFactor Fixes

nonnorm committed:
Remove random indent

nonnorm committed:
Change RegEx

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -8550,8 +8550,7 @@ _EOF_
 			then
-				G_WHIP_BUTTON_CANCEL_TEXT='Skip'
-				G_WHIP_INPUTBOX "Please enter your public IPv6 address. Skip this if you only want to use IPv4, or if you don't have a static IPv6 address." && ipv6=$G_WHIP_RETURNED_VALUE
+				ipv6=$(hostname -I | grep -o "[[:blank:]][[:digit:]:a-f]*")
```
I think this cannot correctly be automated:
- It won't match if there is no IPv4 address assigned, rare but possible.
- We cannot know whether the first IPv6 address in this list is really the public address for the network interface that Tor clients connect through, or from any other network adapter, e.g. for a local network bridge or access point, or if there are multiple IPv6 addresses assigned to the same adapter, which is not uncommon.
- Ah and it includes a leading space, which is I think an issue in below `"ORPort [$ipv6]:$orport"`.

Probably the inputbox could stay, but with the derived IPv6 address as default value, e.g.:
```
for i in $(hostname -I)
do
	[[ i == *.* ]] && continue
	G_WHIP_DEFAULT_ITEM=$i
	break
done
```
If it contains a dot, it can only be an IPv4 address, else must be an IPv6 address. Probably there is an easier solution with grep/sed/mawk 😅.
_____________
I was just looking through the docs to see whether there is a way to leave out the IPv6 address, but indeed it is mandatory to make the server reachable via IPv6. But `[[ $ipv6 ]] && echo 'IPv6Exit 1' >> /etc/tor/torrc` should go into the exit relay section as it is only for exit relays, if I don't misunderstand the wiki: https://gitlab.torproject.org/legacy/trac/-/wikis/TorRelayGuide

nonnorm committed:
Add CookieAuthentication

nonnorm committed:
Add INPUTBOX

nonnorm committed:
Move IPv6Exit

MichaIng committed:
v6.34

+ DietPi-Software | Tor Relay: Re-add "Skip" cancel button to IPv6 inputbox
+ DietPi-Software | Tor Relay: Only allow IPv6 connections if we actually have an IPv6 address to add
